src/dbnode/storage/series/series.go

```diff
@@ -519,7 +520,12 @@ func (s *dbSeries) OnRetrieveBlock(
 	segment ts.Segment,
 ) {
 	s.Lock()
-	defer s.Unlock()
+	shouldUnlock := true
+	defer func() {
+		if shouldUnlock {
+			s.Unlock()
+		}
+	}()
```

Comment:
Could we refactor this to avoid multiple unlock calls?  Perhaps something like:
```go
var (
  b block.DatabaseBlock
  updateList *block.WiredList
)

s.Lock()
defer func() {
  s.Unlock()
  if b != nil && updateList != nil {
    updateList.BlockingUpdate(b) // Update outside of lock
  }
}()
```

Reply:
oh yeah thats better thanks